scenes/enemies/guard/guard.gd

```diff
@@ -0,0 +1,395 @@
+@tool
+class_name Guard extends CharacterBody2D
+## Enemy type that patrols along a path and raises an alert if the player is detected.
+
+## Emitted when the player is detected.
+signal player_detected(player: Node2D)
+
+enum State {
+	## Going along the path.
+	PATROLING,
+	## Player is in sight, it takes some time until the player is detected.
+	DETECTING,
+	## Player was detected.
+	ALERTED,
+	## Player was in sight, going to the last point where the player was seen.
+	INVESTIGATING,
+	## Lost track of player, walking back to the patrol path.
+	RETURNING
```

Comment:
Although the linter doesn't detect it, it is better style to have a trailing comma here. See https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html#trailing-comma
```suggestion
	RETURNING,
```

Reply:
do you know if that could be added to the linter? 🤔

Reply:
The linter is https://github.com/Scony/godot-gdscript-toolkit/wiki/3.-Linter + https://github.com/Scony/godot-gdscript-toolkit/wiki/4.-Formatter . 

I believe that, like Python's `black`, the presence or absence of a trailing comma is something that's treated specially by the formatter. If you have:

```gdscript
enum Foo { A, B }
```

it will stay on one line if it can. If you have a trailing comma:

```gdscript
enum Foo { A, B, }
```

It will always be wrapped with one element per line.

Reply:
I checked the documentation and there's no way to force one style or other.

Reply:
Ohhh I see, and also from what I've checked the formatter isn't configurable (well, only max-length)
https://github.com/Scony/godot-gdscript-toolkit/wiki/4.-Formatter